lib/galaxy/tools/__init__.py

```diff
@@ -2541,7 +2541,7 @@ def iterate(group_inputs, inputs, state_inputs, other_values=None):
             other_values = ExpressionContext( state_inputs, other_values )
             for input_index, input in enumerate( inputs.itervalues() ):
                 # create model dictionary
-                tool_dict = input.to_dict(trans)
+                tool_dict = input.to_dict(trans, other_values=other_values)
```

Comment:
@guerler 

The dynamic fields only work without these changes because until now no tool includes 2 dynamically generated select lists that are associated.

These inputs produce the exception without my changes:

        <param name="input" type="data" format="plyascii,plybinary,vtkascii,vtkbinary" label="Shape">
            <validator type="dataset_ok_validator" />
        </param>
        <param name="field_name" type="text" value="surface_field" label="Surface field name" help="Spaces will be converted to underscore character."/>
        <param name="expression" type="text" value="(x**2 * sin(y*pi) + z**4) * exp(t)" label="Expression" help="Legal variables are x,y,z (point coordinates) and t (time)."/>
        <repeat name="time_points" title="Points in time" min="1" help="Provides snapshots of surface field over time sequence.">
            <param name="time_point" type="float" value="0" label="Time point" help="Floating point number"/>
        </repeat>